src/Uri.php

```diff
@@ -71,6 +71,9 @@ class Uri implements UriInterface
     /** @var string Uri fragment. */
     private $fragment = '';
 
+    /** @var string|null String representation */
+    private $toString = null;
```

Comment:
null is the default value anyway, no need to assign it here.

Reply:
Until typed properties (upcoming PHP 7.4) are used, this is true. After that, the property will be uninitialized, not `null`. I tend to future-proof my code this way, but I'll happily remove the default value if you wish.

Reply:
also please name it `$composedComponents`

Reply:
well it's only unitialized when you add the types to the property. so you need to actively break the code. and this would be relevant for all properties, not just this one. so let's deal with that later.

Reply:
Sure thing! Done.